chrome/content/Overlays.jsm

```diff
@@ -0,0 +1,861 @@
+    return deferredLoad;
+  }
+
+  /**
+   * Load the CSS stylesheet from the given url
+   *
+   * @param {String} url        The url to load from
+   * @return {Element}          An HTML link element for this stylesheet
+   */
+  loadCSS(url) {
+    console.debug(`Loading ${url} into ${this.window.location}`);
+
+    let winUtils = this.window.windowUtils;
+    winUtils.loadSheetUsingURIString(url, winUtils.AUTHOR_SHEET);
+  }
+
+  trueAttribute(obj, attr) {
+    if(!obj || !obj.getAttribute) { return false; }
+  
+    return (obj.getAttribute(attr) == 'true');
+  };
+
+  removeAttribute(obj, attr) {
+    if(!obj || !obj.removeAttribute) { return; }
+    obj.removeAttribute(attr);
+  };
+
+  appendButton(aWindow, palette, node) {
+		if(!node.parentNode) {
+			palette.appendChild(node);
+		}
+		var id = node.id;
+
+		var widget = CustomizableUI.getWidget(id);
+		if(!widget || widget.provider != CustomizableUI.PROVIDER_API) {
+			// this needs the binding applied on the toolbar in order for the widget to be immediatelly placed there,
+			// and since its placements won't be restored until it's created, we have to search for it in all existing areas
+			var areaId = null;
+			var areas = CustomizableUI.areas;
+			for(let area of areas) {
+				// this will throw if called too early for an area whose placements have not been fetched yet,
+				// it's ok because once they are, the widget will be placed in it anyway
+				try { var inArea = CustomizableUI.getWidgetIdsInArea(area); }
+				catch(ex) { continue; }
+
+				if(inArea.indexOf(id) > -1) {
+					if(CustomizableUI.getAreaType(area) != CustomizableUI.TYPE_TOOLBAR) { break; }
+
+					areaId = area;
+					this.tempAppendAllToolbars(aWindow, area);
+					break;
+				}
+			}
+
+			try { CustomizableUI.createWidget(this.getWidgetData(aWindow, node, palette)); }
+			catch(ex) { Cu.reportError(ex); }
+
+			if(areaId) {
+				this.tempRestoreAllToolbars(aWindow, areaId);
+			}
+		}
+
+		else {
+			var placement = CustomizableUI.getPlacementOfWidget(id, aWindow);
+			var areaNode = (placement) ? aWindow.document.getElementById(placement.area) : null;
+			if(areaNode && areaNode.nodeName == 'toolbar' && !areaNode._init) {
+				this.tempAppendToolbar(aWindow, areaNode);
+			}
+
+			try { CustomizableUI.ensureWidgetPlacedInWindow(id, aWindow); }
+			catch(ex) { Cu.reportError(ex); }
+
+			if(areaNode) {
+				this.tempRestoreToolbar(areaNode);
+			}
+		}
+
+		// this.traceBack(aWindow, {
+		// 	action: 'appendButton',
+		// 	node: node
+		// });
+		return node;
+  }
+  
+  tempAppendToolbar(aWindow, node) {
+		if(node.tempAppend) {
+			Cu.reportError('tempAppend already exists!');
+			return;
+		}
+
+		node.tempAppend = {
+			parent: node.parentNode,
+			sibling: node.nextSibling,
+			container: aWindow.document.createElement('box')
+		};
+
+		setAttribute(node.tempAppend.container, 'style', 'position: fixed; top: 4000px; left: 4000px; opacity: 0.001;');
```

Comment:
'setAttribute' is not defined ?

Reply:
Is this one complain an error on you environment?